internal/commands/stack/deploy_supabase.go

```diff
@@ -0,0 +1,308 @@
+	// Check if the network already exists
+	networks, err := exec.Network().GetNetworks(exec.Context())
+	if err != nil {
+		return fmt.Errorf("failed to get networks: %w", err)
+	}
+
+	networkExists := false
+	for _, network := range networks.Networks {
+		if network.Name == networkName {
+			networkExists = true
+			networkID = network.UUID
+			break
+		}
+	}
+
+	// Create the network if it does not exist
+	if !networkExists {
+		fmt.Println("Network does NOT exists, creating:", networkName)
+		var networkCreated bool = false
+
+		for i := 0; i < 10; i++ {
+			// Generate random 10.0.X.0/24 subnet
+			x := rand.Intn(254) + 1
+			cidr := fmt.Sprintf("10.0.%d.0/24", x)
+			fmt.Println("Trying to create network with CIDR:", cidr)
+
+			network, err := exec.Network().CreateNetwork(exec.Context(), &request.CreateNetworkRequest{
+				Name:       networkName,
+				Zone:       location,
+				Router:     "",
+				IPNetworks: []upcloud.IPNetwork{{Address: cidr, DHCP: 1, Family: upcloud.IPAddressFamilyIPv4}},
+			})
+
+			if err != nil {
+				fmt.Printf("Failed to create network %s with CIDR %s: %v\n", networkName, cidr, err)
+				continue
+			} else {
+				fmt.Println("Network created successfully:", network.Name, "with network.UUID", network.UUID)
+				networkCreated = true
+				networkID = network.UUID
+				networkCIDR = cidr
+				break
+			}
+		}
+		if !networkCreated {
+			return fmt.Errorf("failed to create network after 10 attempts")
+		}
+	}
+
+	fmt.Println(("Moving to creating Kubernetes cluster..."))
+	cluster, err := exec.All().CreateKubernetesCluster(exec.Context(), &request.CreateKubernetesClusterRequest{
+		Name:        clusterName,
+		Network:     networkID,
+		Zone:        location,
+		NetworkCIDR: networkCIDR,
+		Labels: []upcloud.Label{
+			{Key: "stacks.upcloud.com/stack", Value: "supabase"},
+			{Key: "stacks.upcloud.com/created-by", Value: "upctl"},
+			{Key: "stacks.upcloud.com/chart-version", Value: "0.1.3"},
+		},
+		NodeGroups: []request.KubernetesNodeGroup{
+			{
+				Name:  "supabase-node-group",
+				Count: 1,
+				Plan:  "2xCPU-4GB",
+			},
+		},
+	})
+
+	if err != nil {
+		return fmt.Errorf("failed to create Kubernetes cluster: %w", err)
+	}
+
+	fmt.Println("Kubernetes cluster created successfully:", cluster.Name, "with UUID", cluster.UUID)
+
+	fmt.Println("Requesting kubeconfig for the cluster...")
+	kubeconfig, err := exec.All().GetKubernetesKubeconfig(exec.Context(), &request.GetKubernetesKubeconfigRequest{
+		UUID: cluster.UUID,
+	})
+
+	if err != nil {
+		return fmt.Errorf("failed to get kubeconfig for cluster %s: %w", cluster.UUID, err)
+	}
+
+	fmt.Println("Kubeconfig received successfully for cluster:", cluster.Name)
+
+	// Create a tmp dir for this deployment
+	chartDir, err := os.MkdirTemp("", fmt.Sprintf("supabase-%s-%s", name, location))
+	if err != nil {
+		return fmt.Errorf("failed to make temp dir for deployment: %w", err)
+	}
+
+	// Save kubeconfig to temp file
```

Comment:
Could we store this in memory for the kubernetes client, so no need for setting the env variable?